consul/data_source_consul_service_health.go

```diff
@@ -16,84 +16,81 @@ func dataSourceConsulServiceHealth() *schema.Resource {
 		Read: dataSourceConsulServiceHealthRead,
 		Schema: map[string]*schema.Schema{
 			// Filter parameters
-			"datacenter": &schema.Schema{
+			"datacenter": {
 				Optional: true,
 				Type:     schema.TypeString,
-				// ForceNew: true,
 			},
-			"name": &schema.Schema{
+			"name": {
 				Required: true,
 				Type:     schema.TypeString,
-				// ForceNew: true,
 			},
-			"near": &schema.Schema{
+			"near": {
 				Optional: true,
 				Type:     schema.TypeString,
-				// ForceNew: true,
 			},
-			"tag": &schema.Schema{
+			"tag": {
 				Optional: true,
 				Type:     schema.TypeString,
-				// ForceNew: true,
 			},
-			"node_meta": &schema.Schema{
+			"node_meta": {
 				Optional: true,
 				Type:     schema.TypeMap,
 				Elem: &schema.Schema{
 					Type: schema.TypeString,
 				},
-				// ForceNew: true,
 			},
-			"passing": &schema.Schema{
+			"passing": {
 				Optional: true,
 				Type:     schema.TypeBool,
-				// ForceNew: true,
-				Default: true,
+				Default:  true,
 			},
-			"wait_for": &schema.Schema{
+			"wait_for": {
+				Optional: true,
+				Type:     schema.TypeString,
+			},
+			"filter": {
```

Comment:
Should the filter have `ForceNew: true` or is that not necessary to make terraform make an api request with the new filter?

Reply:
After a little more thought I am fairly certain its not needed and TF will do the right thing as-is. Getting some confirmation on that point would be great though.

Reply:
Good question, for datasources `ForceNew` is not needed as a request they only implement `Read` they will always make an API call no matter whether their arguments changed. `ForceNew` is only meaningful for resources that implements `Update`, for resources that don't implement `Update` all arguments must have `ForceNew: true`.

Reply:
👍